### PR TITLE
Fix search functionality

### LINKED ISSUE #5: Search Functionality for Blogs and Learnings
**Is your feature request related to a problem? Please describe.**
As content in the "Blogs" and "Learnings" sections grows, it becomes difficult to quickly find specific articles or topics without a dedicated search feature.

**Describe the solution you'd like**
Implement a search bar that allows users to search across both "Blogs" and "Learnings" sections using keywords, titles, or tags. This would improve content discoverability and save users time when looking for specific information.

**Describe alternatives you've considered**
Manually browsing through content is an option, but it’s time-consuming and not practical for large collections of posts.

**Additional context**
A search feature would make the portfolio template more scalable and user-friendly as more content is added.

### PR DESCRIPTION
**Pull request by NitanJana**
## Summary

This PR should close #5  by fixing the searchbar.

## Functionality implemented

- [x] indexes the whole site at build time.
- [x] searchbar resets at route change.
- [x] show only at /blog and /learning.
- [x] fix dev mode.
- [ ] index specific elements instead of whole site.
- [ ] dark mode support/ other visual changes.

### CODE REVIEW

**Comment by NitanJana** [comment]
@yaralahruthik sorry for the conflict, didn't see you merged my previous commit.

File: src/site-config.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ type SiteConfig = {
     name: string;
     url: string;
   }[];
+  enableSearch: boolean;
```
Comment (yaralahruthik) [inline src/site-config.ts:15]: Hmm, I was thinking that maybe we use environment variables for this. That way we can conditionally also handle the build step.

Want to give it a try?

**Review by yaralahruthik — APPROVED** [review]
Looks good to me!

File: src/site-config.ts (hunk 1 above)
Comment (NitanJana) [reply src/site-config.ts:15]: Sure, I think enableSearch: import.meta.env.ENABLE_SEARCH should do the trick.

**Pull request merged**